pkg/operation/botanist/etcdencryption.go

```diff
@@ -0,0 +1,321 @@
+	}
+	_, ok = ecs.Annotations[common.EtcdEncryptionForcePlaintextAnnotationName]
+	return true, ec, ok, nil
+}
+
+// calculateEtcdEncryptionSecretNameInGardenCluster concatenates the shoot name and the suffic for the etcd encryption secret
+func (b *Botanist) calculateEtcdEncryptionSecretNameInGardenCluster() string {
+	return fmt.Sprintf("%s.%s", b.Shoot.Info.Name, common.EtcdEncryptionSecretName)
+}
+
+// readEncryptionConfigurationFromGarden reads the EncryptionConfiguration from the shoot namespace in the seed
+func (b *Botanist) readEncryptionConfigurationFromGarden() (bool, *apiserverconfigv1.EncryptionConfiguration, error) {
+	secretNameInGardenCluster := b.calculateEtcdEncryptionSecretNameInGardenCluster()
+	ecs, err := b.K8sGardenClient.GetSecret(b.Shoot.Info.Namespace, secretNameInGardenCluster)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return false, nil, nil
+		}
+		return false, nil, err
+	}
+	secretData, ok := ecs.Data[common.EtcdEncryptionSecretFileName]
+	if !ok {
+		return true, nil, fmt.Errorf("EncryptionConfiguration in garden cluster (%v) does not contain expected element: %v", secretNameInGardenCluster, common.EtcdEncryptionSecretFileName)
+	}
+	ec, err := encryptionconfiguration.CreateFromYAML(secretData)
+	if err != nil {
+		return true, nil, fmt.Errorf("EncryptionConfiguration in garden cluster (%v) is not consistent: %v", secretNameInGardenCluster, err)
+	}
+	return true, ec, nil
+}
+
+// writeEncryptionConfiguration writes the secret which contains the EncryptionConfiguration to the
+// shoot namespace in the seed cluster as well as to the garden cluster
+func (b *Botanist) writeEncryptionConfiguration(ecYamlBytes []byte, forcePlaintextSecrets bool) error {
+	err := b.writeEncryptionConfigurationSecretToSeed(ecYamlBytes, forcePlaintextSecrets)
+	if err != nil {
+		return err
+	}
+	return b.writeEncryptionConfigurationSecretToGarden(ecYamlBytes)
+}
+
+func (b *Botanist) createEncryptionConfigurationSecret(name string, namespace string, ecYamlBytes []byte, forcePlaintextSecrets bool) *corev1.Secret {
+	secretObj := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			common.EtcdEncryptionSecretFileName: ecYamlBytes,
+		},
+	}
+	if forcePlaintextSecrets {
+		secretObj.ObjectMeta.Annotations = map[string]string{
+			common.EtcdEncryptionForcePlaintextAnnotationName: "true",
+		}
+	}
+	return &secretObj
+}
+
+// writeEncryptionConfigurationSecretToSeed writes the secret which contains the EncryptionConfiguration
+// to the shoot namespace in the seed cluster
+func (b *Botanist) writeEncryptionConfigurationSecretToSeed(ecYamlBytes []byte, forcePlaintextSecrets bool) error {
+	secretObj := b.createEncryptionConfigurationSecret(common.EtcdEncryptionSecretName, b.Shoot.SeedNamespace, ecYamlBytes, forcePlaintextSecrets)
+	if _, err := b.K8sSeedClient.CreateSecretObject(secretObj, true); err != nil {
```

Comment:
With #1113 we try to remove most of these deprecated client functions.
Try using `CreateOrUpdate` from the controller-runtime to create or update the secret object.
(Same for other occurrences, I don't comment on every one separately.)